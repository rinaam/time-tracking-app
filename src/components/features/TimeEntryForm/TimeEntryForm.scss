@import '../../../styles/colors.scss';
@import '../../../styles/spacing.scss';
@import '../../../styles/variables.scss';

.time-entry-wrapper {
  border: $border-primary;
  padding: $spacing-s;
  background-color: $background-white;

  .section {
    margin-bottom: $spacing-s;
    display: flex;

  }
}