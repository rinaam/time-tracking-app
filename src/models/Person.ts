export interface Person {
  id: string;
  type: string;
  attributes: {
    first_name: string;
    last_name: string;
    email: string;
    title: unknown | null;
    joined_at: string;
    last_seen_at: string;
    deactivated_at: unknown | null;
    archived_at: unknown | null;
    role_id: number;
    setting_role_id: unknown | null;
    invited_at: unknown | null;
    is_user: boolean;
    user_id: number;
    tag_list: unknown[];
    avatar_url: unknown | null;
    virtual: boolean;
    custom_fields: unknown | null;
    autotracking: boolean;
    created_at: string;
    placeholder: boolean;
    color_id: unknown | null;
    contact: unknown;
    sample_data: boolean;
    time_unlocked: boolean;
    time_unlocked_on: unknown | null;
    time_unlocked_start_date: unknown | null;
    time_unlocked_end_date: unknown | null;
    time_unlocked_period_id: unknown | null;
    time_unlocked_interval: unknown | null;
    last_activity_at: string;
    two_factor_auth: boolean;
  };
  relationships: {
    organization: {
      data: {
        type: string;
        id: string;
      };
    };
    company: {
      data: {
        type: string;
        id: string;
      };
    };
    subsidiary: {
      data: {
        type: string;
        id: string;
      };
    };
    holiday_calendar: {
      data: {
        type: string;
        id: string;
      };
    };
  };
}
