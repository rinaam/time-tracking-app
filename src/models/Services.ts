export interface ServicesResponseModel {
  data: {
    id: string;
    type: string;
    attributes: {
      name: string;
      position: number;
      deleted_at: unknown | null;
      billable: boolean;
      description: unknown | null;
      tracking_enabled: boolean;
      time_tracking_enabled: boolean;
      expense_tracking_enabled: boolean;
      pricing_type_id: number;
      billing_type_id: number;
      worked_time: number;
      unapproved_time: number;
      billable_time: number;
      estimated_time: number;
      booked_time: number;
      budgeted_time: number;
      profit_margin: number;
      unit_id: number;
      quantity: string;
      discount: unknown | null;
      future_booked_time: number;
      currency: string;
      currency_default: string;
      currency_normalized: string;
      hourly_rate: number;
      hourly_rate_default: number;
      hourly_rate_normalized: number;
      price: number;
      price_default: number;
      price_normalized: number;
      revenue: number;
      revenue_default: number;
      revenue_normalized: number;
      cost: number;
      cost_default: number;
      cost_normalized: number;
      work_cost: number;
      work_cost_default: number;
      work_cost_normalized: number;
      expense_amount: number;
      expense_amount_default: number;
      expense_amount_normalized: number;
      expense_billable_amount: number;
      expense_billable_amount_default: number;
      expense_billable_amount_normalized: number;
      profit: number;
      profit_default: number;
      profit_normalized: number;
      budget_total: number;
      budget_total_default: number;
      budget_total_normalized: number;
      budget_used: number;
      budget_used_default: number;
      budget_used_normalized: number;
      future_revenue: number;
      future_revenue_default: number;
      future_revenue_normalized: number;
      future_budget_used: number;
      future_budget_used_default: number;
      future_budget_used_normalized: number;
      future_cost: number;
      future_cost_default: number;
      future_cost_normalized: number;
      discount_amount: number;
      discount_amount_default: number;
      discount_amount_normalized: number;
    };
    relationships: {
      organization: {
        data: {
          type: string;
          id: string;
        };
      };
      service_type: {
        data: {
          type: string;
          id: string;
        };
      };
      deal: {
        data: {
          type: string;
          id: string;
        };
      };
      person: {
        data: null;
      };
      section: {
        data: null;
      };
    };
  }[];
  included: {
    id: string;
    type: string;
    attributes: {
      name: string;
      category_id: unknown | null;
      archived_at: unknown | null;
      description: null;
    };
    relationships: {
      organization: {
        data: {
          type: string;
          id: string;
        };
      };
    };
  }[];
  links: {
    first: string;
    last: string;
  };
  meta: {
    current_page: number;
    total_pages: number;
    total_count: number;
    page_size: number;
    max_page_size: number;
    organization_features: {
      custom_fields: {
        type: string;
        limit: number;
      };
      custom_fields_per_project: {
        type: string;
        limit: number;
      };
      public_custom_reports: {
        type: string;
        limit: number;
      };
      private_custom_reports: {
        type: string;
        limit: number;
      };
      public_project_reports: {
        type: string;
        limit: number;
      };
      private_project_reports: {
        type: string;
        limit: number;
      };
      scheduling_placeholders: {
        type: string;
        limit: number;
      };
      workflows: {
        type: string;
        limit: number;
      };
      rate_cards: {
        type: string;
        limit: number;
      };
      teams: {
        type: string;
        limit: number;
      };
      recycle_bin: {
        type: string;
        available: boolean;
        limit: number;
      };
      multigrouping: {
        type: string;
        available: boolean;
      };
      client_access_to_budgets: {
        type: string;
        available: boolean;
      };
      subsidiaries: {
        type: string;
        available: boolean;
      };
      time_approvals: {
        type: string;
        available: boolean;
      };
      expenses_approvals: {
        type: string;
        available: boolean;
      };
      timeoff_approvals: {
        type: string;
        available: boolean;
      };
      autotracking: {
        type: string;
        available: boolean;
      };
      forecasting: {
        type: string;
        available: boolean;
      };
      webhooks: {
        type: string;
        available: boolean;
      };
      table_pivoting: {
        type: string;
        available: boolean;
      };
      formula_fields: {
        type: string;
        available: boolean;
      };
      single_sign_on: {
        type: string;
        available: boolean;
      };
      duplicate_tasks: {
        type: string;
        available: boolean;
      };
      billable_time_rounding: {
        type: string;
        available: boolean;
      };
      payment_sync: {
        type: string;
        available: boolean;
      };
      booking_methods: {
        type: string;
        available: boolean;
      };
      import_tasks_csv: {
        type: string;
        available: boolean;
      };
      reactions: {
        type: string;
        available: boolean;
      };
      time_week_view: {
        type: string;
        available: boolean;
      };
      subtasks: {
        type: string;
        available: boolean;
      };
      time_calendar_layout: {
        type: string;
        available: boolean;
      };
      theming: {
        type: string;
        available: boolean;
      };
      remove_branding: {
        type: string;
        available: boolean;
      };
      custom_invoicing_email: {
        type: string;
        available: boolean;
      };
      google_calendar_layout: {
        type: string;
        available: boolean;
      };
      pulse: {
        type: string;
        available: boolean;
      };
      slack: {
        type: string;
        available: boolean;
      };
      task_custom_fields_library: {
        type: string;
        available: boolean;
      };
      timeline_layout: {
        type: string;
        available: boolean;
      };
      comment_visibility: {
        type: string;
        available: boolean;
      };
      time_off_sync: {
        type: string;
        available: boolean;
      };
      time_locking: {
        type: string;
        available: boolean;
      };
      docs: {
        type: string;
        available: boolean;
      };
      scheduling_resource_utilization: {
        type: string;
        available: boolean;
      };
      enforce_two_factor_auth: {
        type: string;
        available: boolean;
      };
      personio_integration: {
        type: string;
        available: boolean;
      };
    };
    settings: {
      systemFlags: {
        setting_type: string;
        value: string;
        updated_at: string;
      };
      vismaEconomicIntegration: {
        setting_type: string;
        value: string;
        updated_at: string;
      };
    };
  };
}
