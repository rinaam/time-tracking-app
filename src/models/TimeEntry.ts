export interface TimeEntry {
  id: string;
  type: string;
  attributes: {
    date: string;
    created_at: string;
    time: number;
    billable_time: number;
    note: string;
    track_method_id: number;
    started_at: unknown | null;
    timer_started_at: unknown | null;
    timer_stopped_at: unknown | null;
    approved: boolean;
    approved_at: string;
    updated_at: string;
    calendar_event_id: unknown | null;
    invoiced: boolean;
    overhead: boolean;
    rejected: boolean;
    rejected_reason: unknown | null;
    rejected_at: unknown | null;
    currency: string;
    currency_default: string;
    currency_normalized: string;
    cost: number;
    cost_default: number;
    cost_normalized: number;
    work_cost: number;
    work_cost_default: number;
    work_cost_normalized: number;
    overhead_cost: number;
    overhead_cost_default: number;
    overhead_cost_normalized: number;
    internal_overhead_cost: number;
    internal_overhead_cost_default: number;
    internal_overhead_cost_normalized: number;
    facility_overhead_cost: number;
    facility_overhead_cost_default: number;
    facility_overhead_cost_normalized: number;
  };
  relationships: {
    organization: {
      data: {
        type: string;
        id: string;
      };
    };
    person: {
      data: {
        type: string;
        id: string;
      };
    };
    service: {
      data: {
        type: string;
        id: string;
      };
    };
    task: {
      data: unknown | null;
    };
    approver: {
      data: {
        type: string;
        id: string;
      };
    };
    updater: {
      data: {
        type: string;
        id: string;
      };
    };
    rejecter: {
      data: unknown | null;
    };
  };
}

export interface TimeEntryResponseModel {
  data: TimeEntry[];
  included: {
    id: string;
    type: string;
    attributes: {
      name: string;
      first_name: string;
      last_name: string;
      email: string;
      title: unknown | null;
      joined_at: string;
      last_seen_at: string;
      deactivated_at: unknown | null;
      archived_at: unknown | null;
      role_id: number;
      setting_role_id: unknown | null;
      invited_at: unknown | null;
      is_user: boolean;
      user_id: number;
      tag_list: unknown[];
      avatar_url: unknown | null;
      virtual: boolean;
      custom_fields: unknown | null;
      autotracking: boolean;
      created_at: string;
      placeholder: boolean;
      color_id: unknown | null;
      contact: unknown;
      sample_data: boolean;
      time_unlocked: boolean;
      time_unlocked_on: unknown | null;
      time_unlocked_start_date: unknown | null;
      time_unlocked_end_date: unknown | null;
      time_unlocked_period_id: unknown | null;
      time_unlocked_interval: unknown | null;
      last_activity_at: string;
      two_factor_auth: boolean;
    };
    relationships: {
      organization: {
        data: {
          type: string;
          id: string;
        };
      };
      company: {
        data: {
          type: string;
          id: string;
        };
      };
      subsidiary: {
        data: {
          type: string;
          id: string;
        };
      };
      holiday_calendar: {
        data: {
          type: string;
          id: string;
        };
      };
    };
  }[];
  links: {
    first: string;
    last: string;
    next: string;
  };
  meta: {
    current_page: number;
    total_pages: number;
    total_count: number;
    page_size: number;
    max_page_size: number;
    organization_features: {
      custom_fields: {
        type: string;
        limit: number;
      };
      custom_fields_per_project: {
        type: string;
        limit: number;
      };
      public_custom_reports: {
        type: string;
        limit: number;
      };
      private_custom_reports: {
        type: string;
        limit: number;
      };
      public_project_reports: {
        type: string;
        limit: number;
      };
      private_project_reports: {
        type: string;
        limit: number;
      };
      scheduling_placeholders: {
        type: string;
        limit: number;
      };
      workflows: {
        type: string;
        limit: number;
      };
      rate_cards: {
        type: string;
        limit: number;
      };
      teams: {
        type: string;
        limit: number;
      };
      recycle_bin: {
        type: string;
        available: boolean;
        limit: number;
      };
      multigrouping: {
        type: string;
        available: boolean;
      };
      client_access_to_budgets: {
        type: string;
        available: boolean;
      };
      subsidiaries: {
        type: string;
        available: boolean;
      };
      time_approvals: {
        type: string;
        available: boolean;
      };
      expenses_approvals: {
        type: string;
        available: boolean;
      };
      timeoff_approvals: {
        type: string;
        available: boolean;
      };
      autotracking: {
        type: string;
        available: boolean;
      };
      forecasting: {
        type: string;
        available: boolean;
      };
      webhooks: {
        type: string;
        available: boolean;
      };
      table_pivoting: {
        type: string;
        available: boolean;
      };
      formula_fields: {
        type: string;
        available: boolean;
      };
      single_sign_on: {
        type: string;
        available: boolean;
      };
      duplicate_tasks: {
        type: string;
        available: boolean;
      };
      billable_time_rounding: {
        type: string;
        available: boolean;
      };
      payment_sync: {
        type: string;
        available: boolean;
      };
      booking_methods: {
        type: string;
        available: boolean;
      };
      import_tasks_csv: {
        type: string;
        available: boolean;
      };
      reactions: {
        type: string;
        available: boolean;
      };
      time_week_view: {
        type: string;
        available: boolean;
      };
      subtasks: {
        type: string;
        available: boolean;
      };
      time_calendar_layout: {
        type: string;
        available: boolean;
      };
      theming: {
        type: string;
        available: boolean;
      };
      remove_branding: {
        type: string;
        available: boolean;
      };
      custom_invoicing_email: {
        type: string;
        available: boolean;
      };
      google_calendar_layout: {
        type: string;
        available: boolean;
      };
      pulse: {
        type: string;
        available: boolean;
      };
      slack: {
        type: string;
        available: boolean;
      };
      task_custom_fields_library: {
        type: string;
        available: boolean;
      };
      timeline_layout: {
        type: string;
        available: boolean;
      };
      comment_visibility: {
        type: string;
        available: boolean;
      };
      time_off_sync: {
        type: string;
        available: boolean;
      };
      time_locking: {
        type: string;
        available: boolean;
      };
      docs: {
        type: string;
        available: boolean;
      };
      scheduling_resource_utilization: {
        type: string;
        available: boolean;
      };
      enforce_two_factor_auth: {
        type: string;
        available: boolean;
      };
      personio_integration: {
        type: string;
        available: boolean;
      };
    };
    settings: {
      systemFlags: {
        setting_type: string;
        value: string;
        updated_at: string;
      };
      vismaEconomicIntegration: {
        setting_type: string;
        value: string;
        updated_at: string;
      };
    };
  };
}
