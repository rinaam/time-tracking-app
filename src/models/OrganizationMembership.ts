import { Person } from './Person';

export interface OrganizationMembership {
  id: string;
  type: string;
  attributes: {
    updated_at: string;
    position: unknown | null;
    tasks_layout_id: number;
    time_reminders: boolean;
    email_notifications: boolean;
    weekly_emails: boolean;
    notification_preference: unknown;
    granular_notification_preferences: unknown;
    preferences: unknown | null;
    tasks_filter_id: unknown | null;
    deals_filter_id: unknown | null;
    projects_filter_id: unknown | null;
    people_filter_id: unknown | null;
    companies_filter_id: unknown | null;
    company_time_filter_id: unknown | null;
    invoices_filter_id: unknown | null;
    budgets_filter_id: unknown | null;
    expenses_filter_id: unknown | null;
    bookings_filter_id: unknown | null;
    payments_filter_id: unknown | null;
    time_reports_filter_id: unknown | null;
    profitability_reports_filter_id: unknown | null;
    salary_reports_filter_id: unknown | null;
    time_entry_reports_filter_id: unknown | null;
    booking_approvals_filter_id: unknown | null;
    expense_approvals_filter_id: unknown | null;
    time_approvals_filter_id: unknown | null;
    docs_filter_id: unknown | null;
    permissions: {
      boards: {
        actions: string[];
        editFields: string[];
      };
      deals: {
        actions: string[];
        showFields: string[];
        editFields: string[];
      };
      discussions: {
        actions: string[];
        editFields: string[];
      };
      entitlements: {
        actions: string[];
        editFields: string[];
      };
      expenses: {
        actions: string[];
        showFields: string[];
        editFields: string[];
      };
      pipelines: {
        actions: string[];
        editFields: string[];
      };
      projects: {
        actions: string[];
        showFields: string[];
        editFields: string[];
      };
      project_assignments: {
        actions: string[];
        editFields: string[];
        conditions: {
          actions: {
            owned: string[];
          };
        };
      };
      salaries: {
        actions: string[];
        showFields: string[];
        editFields: string[];
      };
      services: {
        actions: string[];
        showFields: string[];
        editFields: string[];
      };
      service_type_assignments: {
        actions: string[];
        editFields: string[];
      };
      tasks: {
        actions: string[];
        showFields: string[];
        editFields: string[];
        conditions: {
          actions: unknown;
        };
      };
      task_lists: {
        actions: string[];
        editFields: string[];
      };
      teams: {
        actions: string[];
        editFields: string[];
      };
      time_entries: {
        actions: string[];
        showFields: string[];
        editFields: string[];
        conditions: {
          actions: unknown;
        };
      };
    };
  };
  relationships: {
    tasks_favorite_filter: {
      data: unknown | null;
    };
    deals_favorite_filter: {
      data: unknown | null;
    };
    projects_favorite_filter: {
      data: unknown | null;
    };
    people_favorite_filter: {
      data: unknown | null;
    };
    companies_favorite_filter: {
      data: unknown | null;
    };
    company_time_favorite_filter: {
      data: unknown | null;
    };
    invoices_favorite_filter: {
      data: unknown | null;
    };
    budgets_favorite_filter: {
      data: unknown | null;
    };
    expenses_favorite_filter: {
      data: unknown | null;
    };
    bookings_favorite_filter: {
      data: unknown | null;
    };
    payments_favorite_filter: {
      data: unknown | null;
    };
    time_reports_favorite_filter: {
      data: unknown | null;
    };
    profitability_reports_favorite_filter: {
      data: unknown | null;
    };
    salary_reports_favorite_filter: {
      data: unknown | null;
    };
    time_entry_reports_favorite_filter: {
      data: unknown | null;
    };
    booking_approvals_favorite_filter: {
      data: unknown | null;
    };
    expense_approvals_favorite_filter: {
      data: unknown | null;
    };
    time_approvals_favorite_filter: {
      data: unknown | null;
    };
    docs_favorite_filter: {
      data: unknown | null;
    };
    organization: {
      data: {
        type: string;
        id: string;
      };
    };
    person: {
      data: {
        type: string;
        id: string;
      };
    };
    user: {
      data: {
        type: string;
        id: string;
      };
    };
    timer: {
      data: unknown | null;
    };
    organization_membership_counts: {
      data: {
        type: string;
        id: string;
      };
    };
  };
}

export interface OrganizationMembershipResponseModel {
  data: OrganizationMembership[];
  included: Person[];
  links: {
    first: string;
    last: string;
  };
  meta: {
    current_page: number;
    total_pages: number;
    total_count: number;
    page_size: number;
    max_page_size: number;
    settings: {
      systemFlags: {
        setting_type: string;
        value: string;
        updated_at: string;
      };
      vismaEconomicIntegration: {
        setting_type: string;
        value: string;
        updated_at: string;
      };
    };
  };
}
