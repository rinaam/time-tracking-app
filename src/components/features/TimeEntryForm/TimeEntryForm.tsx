import React, { ChangeEvent, SyntheticEvent, useState } from 'react';
import { ServicesResponseModel } from '../../../models/Services';
import Button from '../../core/Button/Button';
import Dropdown, { DropdownOptions } from '../../core/Dropdown/Dropdown';
import Input from '../../core/Input/Input';
import TextArea from '../../core/TextArea/TextArea';
import './TimeEntryForm.scss';

export interface AddEntryData {
  service: string;
  time: string;
  note: string;
}

interface TimeEntryFormProps {
  services?: ServicesResponseModel;
  onAddEntry: (args: AddEntryData) => void;
}

const mapToDropdownOptions = (services?: ServicesResponseModel): DropdownOptions[] =>
  (services?.data || []).map((service) => ({
    label: service.attributes.name,
    value: service.id,
  }));

const TimeEntryForm: React.FC<TimeEntryFormProps> = ({ services, onAddEntry }) => {
  const [service, setService] = useState('-');
  const [time, setTime] = useState('');
  const [note, setNote] = useState('');

  const resetToInitialState = () => {
    setService('-');
    setTime('');
    setNote('');
  };

  const handleSubmit = (e: SyntheticEvent) => {
    e.preventDefault();

    const formData = {
      service,
      time,
      note,
    };

    onAddEntry(formData);
    resetToInitialState();
  };

  const handleStateChange =
    (fn: React.Dispatch<React.SetStateAction<string>>) => (e: ChangeEvent<any>) => {
      fn(e.currentTarget.value);
    };

  const isButtonDisabled = !note || !time || service === '-';

  return (
    <form data-testid="time-entries-table" onSubmit={handleSubmit} className="time-entry-wrapper">
      <div className="section">
        <Dropdown
          label="Service"
          options={mapToDropdownOptions(services)}
          name="service"
          value={service}
          onChange={handleStateChange(setService)}
        />
        <Input
          label="Time"
          id="time"
          name="time"
          type="number"
          value={time}
          onChange={handleStateChange(setTime)}
        />
      </div>
      <TextArea
        label="Note"
        id="note"
        name="note"
        value={note}
        onChange={handleStateChange(setNote)}
      />
      <Button text="Save" type="submit" disabled={isButtonDisabled} />
    </form>
  );
};

export default TimeEntryForm;
