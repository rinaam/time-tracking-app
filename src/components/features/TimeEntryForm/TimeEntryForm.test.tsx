import React from 'react';
import { render, screen, fireEvent, within } from '@testing-library/react';
import TimeEntryForm from './TimeEntryForm';
import servicesFixture from '../../../tests/fixtures/services';
import { ServicesResponseModel } from '../../../models/Services';

describe('TimeEntryForm', () => {
  it('should render component and match snapshot', async () => {
    const { asFragment } = render(
      <TimeEntryForm services={servicesFixture as ServicesResponseModel} onAddEntry={jest.fn()} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should not submit new entry if button is disabled', async () => {
    const fakeSubmit = jest.fn();
    render(
      <TimeEntryForm services={servicesFixture as ServicesResponseModel} onAddEntry={fakeSubmit} />,
    );

    const button = await screen.findByText('Save');

    fireEvent.click(button);

    expect(fakeSubmit).not.toBeCalled();
  });

  it('should submit new entry if all fields are filled in', async () => {
    const fakeSubmit = jest.fn();
    render(
      <TimeEntryForm services={servicesFixture as ServicesResponseModel} onAddEntry={fakeSubmit} />,
    );

    const service = await screen.findByTestId('test-id-service');
    fireEvent.click(service);

    const serviceItem = await within(service).findByText('Acquiring new clients');

    fireEvent.click(serviceItem);

    const note = await screen.findByTestId('test-id-note');
    fireEvent.change(note, { target: { value: 'note' } });

    const time = await screen.findByTestId('test-id-time');
    fireEvent.change(time, { target: { value: '10' } });

    const button = await screen.findByText('Save');
    fireEvent.click(button);

    expect(fakeSubmit).toBeCalledWith({ note: 'note', service: '1687626', time: '10' });
  });
});
