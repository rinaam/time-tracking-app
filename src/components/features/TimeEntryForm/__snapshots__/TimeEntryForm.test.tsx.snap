// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeEntryForm should render component and match snapshot 1`] = `
<DocumentFragment>
  <form
    class="time-entry-wrapper"
    data-testid="time-entries-table"
  >
    <div
      class="section"
    >
      <div
        class="dropdown-wrapper"
      >
        <label
          class="dropdown-label"
        >
          Service
        </label>
        <div
          class="dropdown-container"
          data-testid="test-id-service"
        >
          <p />
        </div>
      </div>
      <div
        class="input-wrapper"
      >
        <label
          class="input-label"
          for="time"
        >
          Time
        </label>
        <input
          class="input-field"
          data-testid="test-id-time"
          id="time"
          name="time"
          type="number"
          value=""
        />
      </div>
    </div>
    <div
      class="text-wrapper"
    >
      <label
        class="text-label"
        for="note"
      >
        Note
      </label>
      <textarea
        class="text-field"
        data-testid="test-id-note"
        id="note"
        name="note"
      />
    </div>
    <button
      class="btn-wrapper btn-disabled"
      disabled=""
      type="submit"
    >
      Save
    </button>
  </form>
</DocumentFragment>
`;
